src/server/sql/weather_data/create_weather_data_table.sql

```diff
@@ -0,0 +1,12 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+CREATE TABLE IF NOT EXISTS weather_data (
+    weather_location_id INTEGER NOT NULL,
+    start_time TIMESTAMP NOT NULL,
```

Comment:
src/server/sql/reading/create_readings_table.sql has the end_time and reading (temperature) as not null. It avoids issues and maybe makes sense here.  It also has ``	CHECK (start_timestamp < readings.end_timestamp),`` which also makes sense to me.